tests/DCPS/Messenger/publisher.cpp

```diff
@@ -86,12 +86,26 @@ int ACE_TMAIN(int argc, ACE_TCHAR *argv[])
       DDS::PropertySeq& props = part_qos.property.value;
       append(props, "OpenDDS.RtpsRelay.Groups", "Messenger", true);
 
-#if defined(OPENDDS_SECURITY)
+#ifdef OPENDDS_SECURITY
+      // Determine the path to the keys
+      OPENDDS_STRING path_to_tests;
+      const char* dds_root = ACE_OS::getenv("DDS_ROOT");
+      if (dds_root && dds_root[0]) {
+        // Use DDS_ROOT in case we are one of the CMake tests
+        path_to_tests = OPENDDS_STRING("file:") + dds_root + "/tests/";
```

Comment:
This doesn't appear to work with an installed OpenDDS.  We should be supporting that and testing it with the CMake tests.  Is it in one of the CI builds?

Reply:
> This doesn't appear to work with an installed OpenDDS.

No it wouldn't, we would have to have it know where the security files were or maybe copy the files like we copy the rest of the files.

> We should be supporting that and testing it with the CMake tests. Is it in one of the CI builds?

It should get built when the build is doing security, but `run_ci_tests just does` just runs `run_test.pl` without any arguments, so it would only be running inforepo/tcp.

For tick it looks like it builds with installed secure OpenDDS but it doesn't look like I set any of the scoreboard builds to run the cmake tests at all.